extensions/amp-access-laterpay/0.1/laterpay-impl.js

```diff
@@ -437,7 +439,7 @@ export class LaterpayVendor {
     a.href = href;
     a.textContent = this.i18n_.alreadyPurchasedLink;
     this.alreadyPurchasedListener_ = listen(a, 'click', ev => {
-      this.handlePurchase_(ev, href);
+      this.handlePurchase_(ev, href, 'alreadyPurchased');
```

Comment:
Instead of moving the `purchaseType` logic out of `#handlePurchase`, let's add a `data-purchase-type` attribute to this anchor.